### PR TITLE
 loader: Make error_code in evmc_load_and_create() optional

### PR DESCRIPTION
**Pull request by chfast**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 - Changed: [[#293](https://github.com/ethereum/evmc/pull/293)]
   In C++ API `evmc::result::raw()` renamed to `evmc::result::release_raw()`.
+- Changed: [[#311](https://github.com/ethereum/evmc/pull/311)]
+  In `evmc_load_and_create()` the `error_code` is not optional (can be `NULL`).
```
Comment (axic) [inline CHANGELOG.md]: `not` or `now`?

Comment (gumb0) [inline CHANGELOG.md]: "is optional"?

File: test/unittests/test_loader.cpp [hunk 2]
```diff
@@ -301,6 +301,11 @@ TEST_F(loader, load_and_create_failure)
     EXPECT_TRUE(vm == nullptr);
     EXPECT_EQ(ec, EVMC_LOADER_INSTANCE_CREATION_FAILURE);
     EXPECT_STREQ(evmc_last_error_msg(), "creating EVMC instance of failure.vm has failed");
+    EXPECT_TRUE(evmc_last_error_msg() == nullptr);
```
Comment (gumb0) [inline test/unittests/test_loader.cpp]: could be `EXPECT_EQ`

File: CHANGELOG.md (hunk 1 above)
Comment (chfast) [reply CHANGELOG.md]: `now`. Why am I making the same mistake all the time?

File: test/unittests/test_loader.cpp (hunk 2 above)
Comment (chfast) [reply test/unittests/test_loader.cpp]: Because of the GTest issues, this would not compile in VS 2017.

File: include/evmc/loader.h [hunk 3]
```diff
@@ -77,28 +77,36 @@ enum evmc_loader_error_code
...
  * the DLL is not going to be loaded multiple times, but the function will return new VM instance
  * each time.
+ *
+ * @param filename    The null terminated path (absolute or relative) to an EVMC module
+ *                    (dynamically loaded library) containing the VM implementation.
+ *                    If the value is NULL, an empty C-string or longer than the path maximum length
+ *                    the ::EVMC_LOADER_INVALID_ARGUMENT is signaled.
+ * @param error_code  The pointer to the error code. If not NULL the value is set to
+ *                    ::EVMC_LOADER_SUCCESS on success or any other error code as described above.
+ * @return            The pointer to the created VM.
```
Comment (axic) [inline include/evmc/loader.h]: Or `NULL` in case of an error. An `error_code` is supplied in this case.

Comment (chfast) [reply include/evmc/loader.h]: Changed.

**Pull request merged**